src/core/include/openvino/op/scatter_elements_update.hpp

```diff
@@ -44,5 +44,65 @@ class OPENVINO_API ScatterElementsUpdate : public Op {
     bool evaluate_scatter_element_update(const HostTensorVector& outputs, const HostTensorVector& inputs) const;
 };
 }  // namespace v3
+namespace v12 {
+class OPENVINO_API ScatterElementsUpdate : public Op {
```

Comment:
At this point I don't see the benefit of inheriting from v3. Perhaps when the reference implementation is added and the v12 gets the evaluate() method this will be changed. In the past inheritance from the previous versions of the op caused problems while working on the opset and sometimes led to a major refactor with base classes creation so I'd rather keep v3 and v12 separate.